src/deploy/functions/params.ts

```diff
@@ -0,0 +1,342 @@
+              referencedParamValue
+          );
+        }
+        output = output.replace(`{{ params.${matches[i]} }}`, referencedParamValue);
+      }
+    }
+    if (output.includes("{{") || output.includes("}}")) {
+      throw new FirebaseError(
+        "CEL evaluation of non-identity expression '" + from + "' not yet supported"
+      );
+    }
+    return output;
+  }
+  return from;
+}
+
+/**
+ * Resolves a boolean field in a Build to an an actual boolean value.
+ * Fields can be literal or a {{ }} delimited CEL expression referencing param values.
+ * Currently only the CEL identity {{ params.PARAMNAME }} is implemented.
+ */
+export function resolveBoolean(
+  from: boolean | build.Expression<boolean>,
+  paramValues: Record<string, build.Field<string | number | boolean>>
+): boolean {
+  if (typeof from === "string" && /{{ params\.(\w+) }}/.test(from)) {
+    const match = /{{ params\.(\w+) }}/.exec(from);
+    const referencedParamValue = paramValues[match![1]];
+    if (typeof referencedParamValue !== "boolean") {
+      throw new FirebaseError(
+        "Referenced boolean parameter '" +
+          match +
+          "' resolved to non-boolean value " +
+          referencedParamValue
+      );
+    }
+    return referencedParamValue;
+  } else if (typeof from === "string") {
+    throw new FirebaseError("CEL evaluation of expression '" + from + "' not yet supported");
+  }
+  return from;
+}
+
+interface ParamBase<T extends string | number | boolean> {
+  // name of the param. Will be exposed as an environment variable with this name
+  param: string;
+
+  // A human friendly name for the param. Will be used in install/configure flows to describe
+  // what param is being updated. If omitted, UX will use the value of "param" instead.
+  label?: string;
+
+  // A long description of the parameter's purpose and allowed values. If omitted, UX will not
+  // provide a description of the parameter
+  description?: string;
+
+  // Default value. If not provided, a param must be supplied.
+  default?: T | build.Expression<T>;
+
+  // default: false
+  immutable?: boolean;
+}
+
+export interface StringParam extends ParamBase<string> {
+  type: "string";
+
+  // If omitted, defaults to TextInput<string>
+  input?: TextInput<string> | SelectInput<string>;
+}
+
+export interface IntParam extends ParamBase<number> {
+  type: "int";
+
+  // If omitted, defaults to TextInput<number>
+  input?: TextInput<number> | SelectInput<number>;
+}
+
+export interface TextInput<T, Extensions = {}> { // eslint-disable-line
+  type?: "text";
+
+  text:
+    | Extensions
+    | {
+        example?: string;
+      };
+}
+
+interface SelectOptions<T> {
+  // Optional human-facing value for this option (e.g. "US Central (Iowa)" instead of value
+  // "us-central1")
+  label?: string;
+
+  // Actual value of the parameter if this option is selected
+  value: T;
+}
+
+export interface SelectInput<T> {
+  type?: "select";
```

Comment:
why is this resolved? you need to remove the `?`

Reply:
Oops. Dumb. Ack.